card/mod/01_core/set/all/collection.rb

```diff
@@ -57,6 +57,27 @@ def include_item? item
   item_names.map{|name| name.to_name.key}.member? key
 end
 
+def add_item item
+  unless include_item? item
+    self.content="#{self.content}\n#{name}"
+  end
+end
+
+def drop_item item
+  if include_item? item
+    new_names = item_names.reject{ |i| i == item }
+    self.content = new_names.empty? ? '' : new_names.join("\n")
+  end
+end
+
+def insert_item index, name
+  new_names = item_names
+  new_names.delete(name)
+  new_names.insert(index,name)
+  self.content =  new_names.join "\n"
+end
```

Comment:
We have methods like `item_names` and `include_item?`in the all set.  No reason to restrict the other item methods to pointers.